cluster/addons/dns/skydns-rc.yaml.in

```diff
@@ -19,6 +19,9 @@ spec:
         version: v11
         kubernetes.io/cluster-service: "true"
     spec:
+{% if grains['cloud'] is defined and grains['cloud'] in [ 'vsphere', 'photon-controller' ] %}
```

Comment:
> In this case, I don't think the issue is the use of hostPort: I added a "hostNetwork" clause, not a "hostPort" clause.

The last sentence in the best practice quote says "Avoid using hostNetwork, for the same reasons as hostPort." so the advice applies here. I'd love to get someone familar with the DNS subsystem to look at this.